lib/Error.hs

```diff
@@ -24,6 +24,7 @@ data CofferError where
   BadFieldNameError :: BadFieldName -> CofferError
   BadMasterFieldName :: Text -> BadFieldName -> CofferError
   BadEntryTagError :: BadEntryTag -> CofferError
+  InvalidPathSegment :: BackendError err => err -> CofferError
```

Comment:
Making `InvalidPathSegment` a wrapper for `BackendError err => err` can lead to invalid states. E.g., it allows us to do this:

```
throw $ InvalidPathSegment $ ServantError ...
```

We should restrict this constructor like this:

```suggestion
InvalidPathSegment
  :: PathSegment
  -> Text -- ^ Backend-specific error message
  -> CofferError
```

In this case, the second argument `Text` would be this section of the error message:

```
Path segments for Vault KV can only contain the following characters:
'#{kvPathSegmentAllowedCharacters}'
```

Reply:
And then delete the `VaultInvalidPathSegment` constructor

Reply:
done